src/KeyVault/Vault.php

```diff
@@ -49,39 +57,27 @@ public function __construct($vaultName, $accessToken)
      */
     protected function requestApi($method, $apiCall, array $json)
     {
-        $client = new Client(
-            [
-                'base_uri' => $this->keyVault,
-                'timeout' => 2.0,
-            ]
-        );
+        $uri = $this->keyVault.$apiCall;
 
-        try {
-            $result = $client->request(
-                $method,
-                $apiCall,
-                [
-                    'headers' => [
-                        'User-Agent' => 'browser/1.0',
-                        'Accept' => 'application/json',
-                        'Content-Type' => 'application/json',
-                        'Authorization' => 'Bearer '.$this->accessToken,
-                    ],
-                    'json' => $json,
-                ]
-            );
+        $request = $this->requestFactory->createRequest($method, $uri)
+            ->withHeader('User-Agent', 'browser/1.0')
+            ->withHeader('Accept', 'application/json')
+            ->withHeader('Content-Type', 'application/json')
+            ->withHeader('Authorization', 'Bearer '.$this->accessToken);
+
+        $request->getBody()->write(json_encode($json));
```

Comment:
Can't remember where the `$json` comes from but if encoding fails it will send `false` to the end-point?

Reply:
Fixed in 67fdb2b